src/core/core/rpc_message.cpp

```diff
@@ -520,7 +543,17 @@ message_ex* message_ex::create_response()
     msg->to_address = header->from_address;
     msg->io_session = io_session;
 
-    // join point 
+    if (hdr.hdr_type == header_type::hdr_dsn_thrift)
+    {
+#ifdef DSN_ENABLE_THRIFT_RPC
+        thrift_header_parser::add_prefix_for_thrift_response(msg);
+        msg->is_response_adjusted_for_custom_rpc = false;
+#else
+        dassert(false, "get request with thrift header type but thrift rpc is disabled");
```

Comment:
can this be done by prepare_send in thrift message parser? it is quite intrusive putting it here.